packages/tizen_rpc_port/lib/src/stub_base.dart

```diff
@@ -33,13 +33,15 @@ abstract class StubBase {
       return pStub.value;
     });
 
-    _finalizer.attach(this, this);
+    _finalizer.attach(this, <String, StubBase>{'stub': this});
```

Comment:
I think the stream subscription should already be closed when when the attachment (`StubBase`) is destroyed, so this should just be safe:

```patch
@@ -33,14 +33,13 @@ abstract class StubBase {
       return pStub.value;
     });

-    _finalizer.attach(this, this);
+    _finalizer.attach(this, _handle);
   }

   late final rpc_port_stub_h _handle;

-  final Finalizer<StubBase> _finalizer = Finalizer<StubBase>((StubBase stub) {
-    stub.close();
-  });
+  final Finalizer<rpc_port_stub_h> _finalizer = Finalizer<rpc_port_stub_h>(
+      (rpc_port_stub_h handle) => tizen.rpc_port_stub_destroy(handle));

   /// A port name to use when listening for connections.
```

Reply:
Thank you for the advice. I thought it was the same case for ProxyBase, so I fixed both.